manager/state/store/tasks.go

```diff
@@ -225,7 +225,7 @@ func (ti taskIndexerByName) FromObject(obj interface{}) (bool, []byte, error) {
 	}
 
 	// Add the null character as a terminator
-	return true, []byte(strings.ToLower(t.ServiceAnnotations.Name) + "\x00"), nil
+	return true, []byte(strings.ToLower(t.Annotations.Name) + "\x00"), nil
```

Comment:
This needs to handle `t.Annotations.Name` and fallback to `t.ServiceAnnotations.Name` with calculated slot and task id.

cc @aaronlehmann